### PR TITLE
helper-cli: GetPackageLicensesCommand improvements

### PR DESCRIPTION
**Pull request by fviernau**

### CODE REVIEW

File: helper-cli/src/main/kotlin/commands/GetPackageLicensesCommand.kt [hunk 1]
```diff
@@ -101,13 +102,12 @@ class GetPackageLicensesCommand : CliktCommand(
                 .applyAll(nonExcludedLicenseFindings, licenseFindingCurations)
                 .mapNotNull { it.curatedFinding }
 
-            val detectedLicense = curatedFindings.map { it.license.toString() }
-                .distinct().sorted().joinToString(" AND ")
+            val detectedLicense = curatedFindings.toSpdxExpression()
```
Comment (mnonnenmacher) [inline helper-cli/src/main/kotlin/commands/GetPackageLicensesCommand.kt:107]: Commit message nit: `asSequence()`

File: helper-cli/src/main/kotlin/commands/GetPackageLicensesCommand.kt [hunk 2]
```diff
@@ -121,7 +122,7 @@ class GetPackageLicensesCommand : CliktCommand(
 }
 
 private fun Collection<LicenseFinding>.toSpdxExpression(): String =
-    asSequence().map { it.license.toString() }.distinct().sorted().joinToString(" AND ")
+    asSequence().map { it.license.toString() }.distinct().joinToString(" AND ").toSpdx().sort().toString()
```
Comment (mnonnenmacher) [inline helper-cli/src/main/kotlin/commands/GetPackageLicensesCommand.kt]: There is no need to convert to string before concatenating and parsing back into an SPDX expression (could also be applied in the previous commit already):
```
asSequence().map { it.license }.reduce(SpdxExpression::and).sort().toString()
```
(not sure if `asSequence()` still makes sense with this code)

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: helper-cli/src/main/kotlin/commands/GetPackageLicensesCommand.kt (hunk 2 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/GetPackageLicensesCommand.kt]: > (not sure if `asSequence()` still makes sense with this code)

The IDE still suggested it, so I kept it.

File: helper-cli/src/main/kotlin/commands/GetPackageLicensesCommand.kt [hunk 3]
```diff
@@ -121,7 +122,7 @@ class GetPackageLicensesCommand : CliktCommand(
 }
 
 private fun Collection<LicenseFinding>.toSpdxExpression(): String =
-    asSequence().distinct().sorted().joinToString(" AND ")
+    asSequence().distinct().joinToString(" AND ").toSpdx().sort().toString()
```
Comment (mnonnenmacher) [inline helper-cli/src/main/kotlin/commands/GetPackageLicensesCommand.kt]: You are now converting the `LicenseFinding`s to strings and still not use `reduce()`:
```
asSequence().map { it.license }.distinct().reduce(SpdxEpxression::and).sort().toString()
```

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: helper-cli/src/main/kotlin/commands/GetPackageLicensesCommand.kt (hunk 3 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/GetPackageLicensesCommand.kt]: I've added the reduce in a separate commit now.

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**